src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -0,0 +1,95 @@
+(ns puppetlabs.jdbc-util.pool
+  (:require [clojure.set :as set]
+            [clojure.tools.logging :as log])
+  (:import [com.zaxxer.hikari HikariConfig HikariDataSource]
+           java.io.Closeable
+           java.sql.SQLTransientConnectionException
+           javax.sql.DataSource))
+
+(defn connection-pool
+  [^HikariConfig config]
+  (let [ds (HikariDataSource. config)]
+    {:datasource ds}))
+
+(defn- set-option
+  [^HikariConfig config option value]
+  (case option
+    :username (.setUsername config value)
+    :password (.setPassword config value)
+    :data-source-class-name (.setDataSourceClassName config value)
+    :jdbc-url (.setJdbcUrl config value)
+    :driver-class-name (.setDriverClassName config value)
+    :auto-commit (.setAutoCommit config value)
+    :connection-timeout (.setConnectionTimeout config value)
+    :idle-timeout (.setIdleTimeout config value)
+    :max-lifetime (.setMaxLifetime config value)
+    :connection-test-query (.setConnectionTestQuery config value)
+    :minimum-idle (.setMinimumIdle config value)
+    :maximum-pool-size (.setMaximumPoolSize config value)
+    :metric-registry (.setMetricRegistry config value)
+    :health-check-registry (.setHealthCheckRegistry config value)
+    :pool-name (.setPoolName config value)
+    (throw (IllegalArgumentException. (format "%s is not a recognized HikariCP option" (str option))))))
+
+(defn options->hikari-config
+  [options]
+  (let [config (HikariConfig.)]
+    (doseq [[option value] options]
```

Comment:
Both of those values have non-zero defaults in Hikari (idleTimeout = 10 minutes, maxLifetime = 30 minutes) so setting one without the other is fine. Hikari will adjust them for you if you've made idletimeout >= maxlifetime, and it'll log that it's making the adjustment unlike the puppetdb code which just silently drops user configuration. You'd be better off removing the conditions.